tests/inputs/lhc_rdts/omc3_helpers.py

```diff
@@ -0,0 +1,145 @@
+import logging
+from pathlib import Path
+
+import tfs
+
+from omc3.hole_in_one import hole_in_one_entrypoint
+from omc3.optics_measurements.constants import RDT_FOLDER
+from tests.inputs.lhc_rdts.rdt_constants import (
+    ANALYSIS_DIR,
+    DATA_DIR,
+    FREQ_OUT_DIR,
+    NORMAL_RDTS3,
+    NORMAL_RDTS4,
+    SKEW_RDTS3,
+    SKEW_RDTS4,
+    LHC_RDTS_TEST_DIR,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+def filter_out_BPM_near_IPs(df: tfs.TfsDataFrame) -> tfs.TfsDataFrame:
+    """Filter the DataFrame to include only BPMs."""
+    return df.filter(regex=r"^BPM\.[1-9][0-9].", axis="index")
+
+
+def get_file_suffix(beam: int) -> str:
+    """Return the file suffix for the test files based on beam and order."""
+    assert beam in [1, 2], "Beam must be 1 or 2"
+    return f"b{beam}"
+
+
+def get_rdts() -> list[str]:
+    """Return the all the RDTs."""
+    return NORMAL_RDTS3 + SKEW_RDTS3 + NORMAL_RDTS4 + SKEW_RDTS4
+
+
+def get_tbt_name(beam: int, sdds: bool = True) -> str:
+    """Return the name of the TBT file for the given test parameters."""
+    return f"tbt_data_{get_file_suffix(beam)}.{'sdds' if sdds else 'tfs.bz2'}"
+
+
+def get_model_dir(beam: int) -> Path:
+    """Return the model directory for the given test parameters."""
+    return LHC_RDTS_TEST_DIR / f"model_{get_file_suffix(beam)}"
+
+
+def get_max_rdt_order(rdts: list[str]) -> int:
+    """Return the maximum order of the RDTs."""
+    return max(sum(int(num) for num in rdt.split("_")[0][1:]) for rdt in rdts)
+
+
+def get_output_dir(tbt_name: str, output_dir: Path = None) -> Path:
+    """Return the output directory for the given TBT, and create it if it does not exist."""
+    if output_dir is None:
+        output_dir = ANALYSIS_DIR / f"{tbt_name.split('.')[0]}"
+        output_dir.mkdir(exist_ok=True)
+    return output_dir
+
+
+def get_rdt_type(rdt: str) -> tuple[str, str]:
+    rdt_as_list = [int(num) for num in rdt.split("_")[0][1:]]
+    is_skew = (rdt_as_list[2] + rdt_as_list[3]) % 2 == 1
+    order = sum(rdt_as_list)
+    return "skew" if is_skew else "normal", "octupole" if order == 4 else "sextupole"
+
+
+def get_rdt_paths(rdts: list[str], output_dir: Path) -> dict[str, Path]:
+    """Return a dictionary of RDTs and their corresponding file paths."""
+    # Is there a method for this in the OMC3 source code? (jgray 2024)
+    rdt_paths = {}
+    for rdt in rdts:
+        is_skew, order_name = get_rdt_type(rdt)
+        rdt_paths[rdt] = output_dir / f"{RDT_FOLDER}/{is_skew}_{order_name}/{rdt}.tfs"
+    return rdt_paths
+
+
+def get_tunes(output_dir: Path) -> list[float]:
+    optics_file = output_dir / "beta_amplitude_x.tfs"
+    headers = tfs.reader.read_headers(optics_file)
+    tunes = [headers["Q1"], headers["Q2"]]
+    return tunes
+
+
+def get_rdts_from_harpy(
```

Comment:
not really the right name for this function, I think.
`get_rdts_from_optics_analysis` maybe?